templates/csharp/message-extension-action/Action/ActionApp.cs.tpl

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using AdaptiveCards;
+using Newtonsoft.Json.Linq;
+
+namespace {{SafeProjectName}}.Action;
+
+public class ActionApp : TeamsActivityHandler
+{ 
+    private readonly string _adaptiveCardFilePath = Path.Combine(".", "Resources", "helloWorldCard.json");
+    // Action.
+    protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionSubmitActionAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
+    {
+        // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+        var adaptiveCardJson = await System.IO.File.ReadAllTextAsync(_adaptiveCardFilePath, cancellationToken);
```

Comment:
From my view, create adaptive card from file template is OK, maybe using embedded constructor is a more common way.
Some reference https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/tab-adaptive-cards/csharp/TabWithAdpativeCardFlow/Helpers/CardHelper.cs